Purchases/Networking/Backend.swift

```diff
@@ -632,6 +652,32 @@ private extension Backend {
         }
     }
 
+    func add(createAliasCallback: PostRequestResponseHandler?, key: String) -> CallbackCacheStatus {
+        return callbackQueue.sync { [self] in
+            var callbacksForKey = createAliasCallbacksCache[key] ?? []
+            let cacheStatus: CallbackCacheStatus = !callbacksForKey.isEmpty
+                ? .addedToExistingInFlightList
+                : .firstCallbackAddedToList
+
+            callbacksForKey.append(createAliasCallback)
+            createAliasCallbacksCache[key] = callbacksForKey
+            return cacheStatus
+        }
+    }
+
+    func add(identifyCallback: @escaping IdentifyResponseHandler, key: String) -> CallbackCacheStatus {
+        return callbackQueue.sync { [self] in
+            var callbacksForKey = identifyCallbacksCache[key] ?? []
+            let cacheStatus: CallbackCacheStatus = !callbacksForKey.isEmpty
+                ? .addedToExistingInFlightList
+                : .firstCallbackAddedToList
+
+            callbacksForKey.append(identifyCallback)
+            identifyCallbacksCache[key] = callbacksForKey
+            return cacheStatus
+        }
+    }
```

Comment:
I was going to say we could create a function that takes a `[String: [T]]`  (identifyCallbacksCache  or createAliasCallbacksCache or offeringsCallbacksCache...), the callback itself and returns the cacheStatus. We could remove some duplicated code. But it will go away when we move this logic somewhere else (as you mentioned in that story you created).

Your call

Reply:
> I was going to say we could create a function that takes a `[String: [T]]` (identifyCallbacksCache or createAliasCallbacksCache or offeringsCallbacksCache...), the callback itself and returns the cacheStatus. We could remove some duplicated code. But it will go away when we move this logic somewhere else (as you mentioned in that story you created).
> 
> Your call

yeah i like that a lot, i figured that kind of refactor could be in the ticket as well. i'll add your note to the tickets